src/features/chat/Chat.tsx

```diff
@@ -0,0 +1,263 @@
+import routes from '../../common/constants/routes.json'
+import { useHistory } from 'react-router-dom'
+import { ChatItem, ChatItemProps } from './ChatItem'
+import {
+  ChatMessage,
+  ChatMessageProps,
+  DIR_RECV,
+  DIR_SENT,
+} from './ChatMessage'
+import editIcon from '../../common/assets/icons/ico-edit.svg'
+import chatMenuIcon from '../../common/assets/icons/ico-chat-menu.svg'
+import msgEmojiIcon from '../../common/assets/icons/ico-chatmsg-emoji.svg'
+import msgAttachIcon from '../../common/assets/icons/ico-chatmsg-link.svg'
+import msgSendIcon from '../../common/assets/icons/ico-chatmsg-send.svg'
+import chatCloseIcon from '../../common/assets/icons/ico-close2.svg'
+import styles from './Chat.module.css'
+import { mockChats, curUser } from './mock-data'
+
+export type LocationState = {
+  param: string
+}
+
+export default function Chat(): JSX.Element {
+  const [newMsgPlaceholder, setNewMsgPlaceholder] = useState(true)
+  const [newMsg, setNewMsg] = useState('')
+  const [activeChatId, setActiveChatId] = useState<number>(mockChats[0].id)
+  const [activeChat, setActiveChat] = useState<ChatItemProps>(mockChats[0])
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [chats, setChats] = useState<ChatItemProps[]>(mockChats) // DEBUG: init state
+
+  /*
+  const location = useLocation<LocationState>()
+
+  useEffect(() => {
+    if (location.state) {
+      setParam(location.state?.param)
+    }
+  }, [location])
+  */
+
+  const endRef = useRef<HTMLDivElement>(null)
+  const scroll2End = () => {
+    endRef.current?.scrollIntoView({ behavior: 'smooth' })
+  }
+
+  const onNewMsgFocus = () => {
+    setNewMsgPlaceholder(false)
+  }
+
+  const onNewMsgBlur = () => {
+    setNewMsgPlaceholder(newMsg.length > 0 ? false : true)
+  }
+
+  const onNewMsgChange = (ev: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setNewMsg(ev.currentTarget.value)
+  }
+
+  /**
+   * Click by active chat menu button "..."
+   */
+  const onActiveChatMenu = () => {
+    // TODO:
+    console.log('show/hide active chat menu')
+  }
+
+  /**
+   * Click by edit button (pencil icon)
+   */
+  const onChatEdit = () => {
+    // TODO:
+    console.log('edit chat')
+  }
+
+  /**
+   * Click by chat in chats' list (select another chat)
+   * @param idx
+   */
+  const onSelectChat = (id: number) => {
+    // TODO:
+
+    const active = chats.find(item => item.id === id)
+    if (active) {
+      setActiveChatId(id)
+      setActiveChat(active)
+    }
+  }
+
+  const onSendMsg = () => {
+    const val = newMsg.replace(/\s{1,}$/, '')
+    if (val.length === 0) {
+      setNewMsg('')
+      return
+    }
+    // TODO:
+
+    // DEBUG
+    const tmp: ChatMessageProps = {
+      text: newMsg,
+      sender: curUser,
+      date: new Date(),
+      unread: true,
+      direction: DIR_SENT,
+    }
+
+    if (activeChat) {
+      activeChat.messages.push(tmp)
+    }
+    setNewMsg('')
+  }
+
+  const onEmoji = () => {
+    // TODO:
```

Comment:
When function contains just a console log - no need to add // TODO: as it is already clear that function is not implemented